### PR TITLE
Reimplement Light.copy to get around VTK8.1 bug

### PR DESCRIPTION
**Pull request by adeak**
### Overview

The original `Light.copy()` implementation relied on `vtkLight.DeepCopy()` and `vtkLight.ShallowClone()`. Since the former was buggy in VTK < 8.2 we now reimplement it from scratch.

I changed so that we always copy manually. I figured it might not be worth the complexity to branch based on VTK version (especially considering testability).

### CODE REVIEW

**Comment by akaszynski** [comment]
> I changed so that we always copy manually. I figured it might not be worth the complexity to branch based on VTK version (especially considering testability).

Considering there's still an issue with the copy method for `vtkLight`, I think it's best to implement entirely within `pyvista`.  Saves us from additional version testing, which will only get more complicated with every VTK release.

**Review by akaszynski — APPROVED** [review]
Approved with one minor suggestion.

**Comment by akaszynski** [comment]
Also, @adeak, do you want to be added to the developers team so you can merge after approval?

**Comment by adeak** [comment]
> Approved with one minor suggestion.

Sorry, I can't find the suggestion.

> do you want to be added to the developers team so you can merge after approval?

If it's not too much hassle for you to keep merging my PRs then I'm good, thank you :)

File: pyvista/plotting/lights.py
Comment (akaszynski) [inline pyvista/plotting/lights.py:864]:
```suggestion
        new_light.transform_matrix = self.transform_matrix
```

**Comment by akaszynski** [comment]
Added the suggestion.  It's completely optional as it's style, but it means we avoid using a temporary variable while still following PEP 20.

**Comment by adeak** [comment]
@akaszynski  OK, but then we need to change the other branch too. I've committed what I think is the only alternative.I leave it up to you whether to include or drop that commit, I'm OK with both variants.

**Comment by akaszynski** [comment]
> OK, but then we need to change the other branch too. I've committed what I think is the only alternative. I leave it up to you whether to include or drop that commit, I'm OK with both variants.

We're good as is; it's a trivial change.

**Pull request merged**
